core/server/master/src/main/java/alluxio/master/file/meta/UfsSyncPathCache.java

```diff
@@ -89,12 +90,14 @@ public boolean shouldSyncPath(String path, long intervalMs) {
 
```

Comment:
I think here, we can add something like 
```
SyncTime lastParentSync = mCache.getIfPresent(PathUtils.getParent(path));
if (isGetStatus && !shouldSyncInternal(lastParentSync, intervalMs, false)) {
   return false;
}
```
basically if we are only getting the status and the parent has been synced, then we return false.
The code path should remain the same for listStatus, because we can't optimize it until we know if it is a dir or file.